### PR TITLE
[test] Remove Monch from supported systems

### LINKED ISSUE #489: Remove Monch from supported systems
Since Monch has been taken out of service, its references in the test cases should be removed. The ***MonchAcceptance tests***, on the other hand, should be kept, since they can be reused (latest example being Fulen).

### PR DESCRIPTION
**Pull request by vkarak**
- Monch-specific tests are commented as obsolete.

Fixes #489.

### CODE REVIEW

**Comment by vkarak** [comment]
@jenkins-cscs retry dom kesch

File: cscs-checks/apps/gromacs/gromacs_check.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ def __init__(self):
         }
 
 
+# FIXME: This test is obsolete; it is kept only for reference.
```
Comment (victorusu) [inline cscs-checks/apps/gromacs/gromacs_check.py]: should we comment the valid systems in this file?

File: cscs-checks/libraries/math/scalapack_compile_run.py [hunk 2]
```diff
@@ -67,16 +67,14 @@ def scalapack_sanity(number1, number2, expected_value):
         ])
 
 
+# FIXME: This test is obsolete; it is kept only for reference.
```
Comment (victorusu) [inline cscs-checks/libraries/math/scalapack_compile_run.py]: should we comment the valid systems in this file?

File: cscs-checks/microbenchmarks/stream/stream.py [hunk 3]
```diff
@@ -11,8 +11,7 @@ def __init__(self):
         self.exclusive_access = True
         # All available systems are supported
         self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
-                              'kesch:cn', 'kesch:pn', 'leone:normal',
-                              'monch:compute']
+                              'kesch:cn', 'kesch:pn', 'leone:normal']
```
Comment (victorusu) [inline cscs-checks/microbenchmarks/stream/stream.py]: we are removing the system, but not the performance reference number.
Is there any reason why? Do we want to keep for future reference?

File: cscs-checks/apps/gromacs/gromacs_check.py (hunk 1 above)
Comment (vkarak) [reply cscs-checks/apps/gromacs/gromacs_check.py]: I was thinking that these tests will be replaced by your versions, so, I've just put a `FIXME` tag.

File: cscs-checks/microbenchmarks/stream/stream.py (hunk 3 above)
Comment (vkarak) [reply cscs-checks/microbenchmarks/stream/stream.py]: I thought we wanted them for reference, don't we?

**Pull request merged**